### PR TITLE
Move deploy to GitHub actions

### PR DESCRIPTION
**Pull request by pezholio**
This adds a new Github action to deploy the infrastructure, and removes the old Travis setup. 

As well as building, tagging and pushing the Docker container, I've added two other jobs to the action - one that runs on develop to deploy to staging and one that runs on master to deploy to prod. Each one runs the same scripts, but with different env vars. I've added all the env vars as secrets in the repo, so this _should_ all work. 

I'd appreciate a review from someone who understands the infrastructure code a bit more than me though, as I haven't really touched it.

### CODE REVIEW

**Comment by pezholio** [comment]
[Look into moving to GitHub actions](https://trello.com/c/PP6NjrA2/1332-look-into-moving-to-github-actions)

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          TF_VAR_skylight_env: ${{ secrets.SKYLIGHT_ENV }}
+          TF_VAR_skylight_enable_sidekiq: ${{ secrets.SKYLIGHT_ENABLE_SIDEKIQ }}
+          TF_VAR_domain: ${{ secrets.STAGING_DOMAIN }}
+          TF_VAR_papertrail_destination: ${{ secrets.STAGING_PAPERTRAIL_DESTINATION }}
+          TF_VAR_google_tag_manager_container_id: ${{ secrets.STAGING_GOOGLE_TAG_MANAGER_CONTAINER_ID }}
+          TF_VAR_google_tag_manager_environment_auth: ${{ secrets.STAGING_GOOGLE_TAG_MANAGER_ENVIRONMENT_AUTH }}
+          TF_VAR_google_tag_manager_environment_preview: ${{ secrets.STAGING_GOOGLE_TAG_MANAGER_ENVIRONMENT_PREVIEW }}
+        run: |
+          script/deploy-terraform
+        if: ${{ github.ref == "refs/heads/staging" }}
```
Comment (leeky) [inline .github/workflows/deploy.yml:43]: We don't have a `staging` branch. Shouldn't this be `develop` ?

Comment (pezholio) [reply .github/workflows/deploy.yml:43]: D'oh, of course!

File: script/deploy-terraform [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+if [ ! -e ~/.tfenv ]
+then
+git clone https://github.com/tfutils/tfenv.git ~/.tfenv
+fi
+export PATH="$HOME/.tfenv/bin:$PATH"
+tfenv install
+
+../install-terraform-provider-for-cf.sh
+
+# CF_PASSWORD, CF_USER, AWS_SECRET_ACCESS_KEY and AWS_ACCESS_KEY_ID
```
Comment (rjw1) [inline script/deploy-terraform:27]: Is CF_USER set anywhere?

**Review by rjw1 — APPROVED** [review]
The changes look good. I have a query about if one of the env vars is set anywhere.
If we are redoing deployment do we want to look at environments stuff in GitHub. 
https://github.blog/changelog/2020-12-15-github-actions-environments-environment-protection-rules-and-environment-secrets-beta/
Given its beta maybe not.

File: script/deploy-terraform (hunk 2 above)
Comment (pezholio) [reply script/deploy-terraform:27]: Oh good catch 👍

**Pull request merged**